### PR TITLE
Changelog and upgrade guide

### PR DESCRIPTION
**Pull request by ralphtheninja**
Closes https://github.com/Level/levelup/issues/540, https://github.com/Level/levelup/issues/546

### CODE REVIEW

**Comment by ralphtheninja** [comment]
Pheeew. Took me three days to write this 😄

**Comment by vweevers** [comment]
> Pheeew. Took me three days to write this 😄

Mad respect :) I'll review later today.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,317 +1,853 @@
...
+# Changelog
+
+## [Unreleased]
+
+### Added
+* Add 9 to Travis (@ralphtheninja)
+
+### Changed
+* Update `browserify` (@ralphtheninja)
+* Update `leveldown` (@ralphtheninja)
```
Comment (vweevers) [inline CHANGELOG.md]: If these are major updates, maybe we should mention the version (same for similar entries below).

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,317 +1,853 @@
...
+### Fixed
+* Restore support for node 4 (@farskipper)
+
+## [2.0.0] - 2017-10-10
+
+### Added
+* Add default export (@zixia)
+* Add test for 'key of queued operation is not serialized' (@vweevers)
+* Add test for 'value of queued operation is not serialized' (@vweevers)
+* Add test for JSON encoding with stream (@vweevers)
```
Comment (vweevers) [inline CHANGELOG.md]: I prefer test as a verb:

- Test that key and value of queued operation are not serialized
- Test JSON encoding with stream

Is shorter, and less monotone between the other "Add x" lines.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,317 +1,853 @@
...
+* Remove `destroy()` (@ralphtheninja)
+* Remove `repair()` (@ralphtheninja)
+* Remove `getProperty()` (@ralphtheninja)
+* Remove `.errorIfExists` (@ralphtheninja)
+* Remove `.createIfMissing` (@ralphtheninja)
+* Remove `.compression` (@ralphtheninja)
+* Remove `.cacheSize` (@ralphtheninja)
+* Remove `.sync` (@ralphtheninja)
+* Remove `.fillCache` (@ralphtheninja)
+* Remove optional `leveldown` (pass in as `options.db`) (@ralphtheninja)
```
Comment (vweevers) [inline CHANGELOG.md]: Did you mean "Passed in as"?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,317 +1,853 @@
...
+* Remove `options` parameter from `maybeError` (not used) (@ralphtheninja)
```
Comment (vweevers) [inline CHANGELOG.md]: "Remove unused `options` parameter"

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,317 +1,853 @@
...
+* Remove `browser` field from `package.json` (@ralphtheninja)
+* Remove 0.12 and 4 from Travis (@juliangruber)
+* Remove `isDefined` from `lib/util.js` (not used) (@ralphtheninja)
```
Comment (vweevers) [inline CHANGELOG.md]: "Remove unused `isDefined` .."

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,317 +1,853 @@
...
+* Replace `tap` with `tape` + `faucet` (@ralphtheninja)
+* Refactor read streams using `level-iterator-stream` and `level-codec` (@juliangruber)
+
+## [1.0.0-2] - 2015-04-30
+
+### Changed
+* Refactor ltgt encoding (@juliangruber)
+
+### Fixed
+* Fix readStream *AsBuffer options (@juliangruber)
```
Comment (vweevers) [inline CHANGELOG.md]: Escape the asterisk

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,317 +1,853 @@
...
+* README: 'arbitrary data object' => 'of any type' (@vweevers)
+* README: reduce 'supported platforms' section (@vweevers)
+* README: rewrite intro and relationship with leveldown (@vweevers)
+* README: cleanup (@vweevers)
+* README: fix bad async code example (@ralphtheninja)
+* Update `deferred-leveldown` (@ralphtheninja)
 
-## 0.10.0 @ Jun 14 2013
+### Removed
+* Remove typings and Typescript tests (@MeirionHughes)
```
Comment (vweevers) [inline CHANGELOG.md]: Maybe: "Remove unstable typings and .." to make it clear why they were removed

File: CHANGELOG.md (hunk 1 above)
Comment (ralphtheninja) [reply CHANGELOG.md]: Updating. Maybe in the future we should actually write the version for all dependencies, even if it's only a minor or a patch.

File: CHANGELOG.md (hunk 3 above)
Comment (ralphtheninja) [reply CHANGELOG.md]: Don't know what I mean here, basically took the commit message as is.

Comment (ralphtheninja) [reply CHANGELOG.md]: I think I'll just remove the `(pass in as ..)` completely

File: CHANGELOG.md (hunk 6 above)
Comment (ralphtheninja) [reply CHANGELOG.md]: Aaaah, this is why the diff looked weird in github.

**Comment by ralphtheninja** [comment]
@vweevers Just merge if you're happy with the update.

**Pull request merged**